e2e/ephemeral_check.go

```diff
@@ -0,0 +1,260 @@
+package e2e
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+
+	snapapi "github.com/kubernetes-csi/external-snapshotter/client/v4/apis/volumesnapshot/v1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+//go:embed testdata/ephemeral_tests/pod-ephemeral-volume-device.yaml
+var podEphemeralBlockYAMLTemplate string
+
+//go:embed testdata/ephemeral_tests/pod-ephemeral-volume-mount.yaml
+var podEphemeralFSYAMLTemplate string
+
+//go:embed testdata/ephemeral_tests/ephemeral-volume-snapshot.yaml
+var ephemeralVolumeSnapshotYAMLTemplate string
+
+//go:embed testdata/ephemeral_tests/ephemeral-clone.yaml
+var ephemeralPvcCloneYAMLTemplate string
+
+//go:embed testdata/ephemeral_tests/ephemeral-snapshot-restore.yaml
+var ephemeralPvcSnapshotRestoreYAMLTemplate string
+
+//go:embed testdata/ephemeral_tests/pod-volume-mount-template.yaml
+var podFSYAMLTemplate string
+
+//go:embed testdata/ephemeral_tests/pod-volume-device-template.yaml
+var podBlockYAMLTemplate string
+
+func ephemeralTest() {
+	Describe("Ephemeral Volume Tests", func() {
+		var pvc *k8sv1.PersistentVolumeClaim = &k8sv1.PersistentVolumeClaim{}
+		var ephemeralPod *k8sv1.Pod
+		var snapshot *snapapi.VolumeSnapshot
+		var clonePvc *k8sv1.PersistentVolumeClaim
+		var clonePod *k8sv1.Pod
+		var restorePvc *k8sv1.PersistentVolumeClaim
+		var restorePod *k8sv1.Pod
+		var err error
+		ctx := context.Background()
+
+		Context("Create ephemeral pod and volume", func() {
+			It("Tests ephemeral volume operations for VolumeMode=Filesystem", func() {
+
+				By("Creating a pod with generic ephemeral volume")
+				podVolumeMountYaml := fmt.Sprintf(podEphemeralFSYAMLTemplate, "ephemeral-filepod", testNamespace, storageClass)
+				ephemeralPod, err = getPod(podVolumeMountYaml)
+				err = crClient.Create(ctx, ephemeralPod)
+				Expect(err).To(BeNil())
+
+				By("PVC should be bound")
+				Eventually(func() bool {
+					err := crClient.Get(ctx, types.NamespacedName{Name: "ephemeral-filepod-generic-ephemeral-volume", Namespace: testNamespace}, pvc)
+					return err == nil && pvc.Status.Phase == k8sv1.ClaimBound
+				}, timeout, interval).Should(BeTrue())
+				fmt.Printf("PVC %s is bound\n", pvc.Name)
+
+				By("Pod should be running")
+				Eventually(func() bool {
+					err = crClient.Get(ctx, types.NamespacedName{Name: ephemeralPod.Name, Namespace: testNamespace}, ephemeralPod)
+					return err == nil && ephemeralPod.Status.Phase == k8sv1.PodRunning
+				}, timeout, interval).Should(BeTrue())
+				fmt.Printf("Pod %s is running\n", ephemeralPod.Name)
+
+				By("Creating a Snapshot of the pvc")
```

Comment:
Write some data in the parent pvc and verify it later on cloned/restored volumes?

Reply:
There's another PR doing this check. https://github.com/red-hat-storage/lvm-operator/pull/235
Will modify once that gets merged.